AmplifyPlugins/API/AWSAPICategoryPluginFunctionalTests/GraphQLModelBased/GraphQLScalarTests.swift

```diff
@@ -0,0 +1,194 @@
+              case .success(let deletedModel) = mutateModel(request: .delete(updatedContainer)),
+              case .success(let emptyModel) = queryModel(request: .get(ScalarContainer.self, byId: container.id)) else {
+            XCTFail("Failed to test CRUD operations")
+            return
+        }
+        XCTAssertEqual(createdModel, container)
+        XCTAssertEqual(updatedModel, updatedContainer)
+        XCTAssertEqual(queriedModel, updatedContainer)
+        XCTAssertEqual(deletedModel, updatedContainer)
+        XCTAssertNil(emptyModel)
+    }
+
+    func testListIntContainer() throws {
+        let container = ListIntContainer(
+            test: 1,
+            nullableInt: 1,
+            intList: [],
+            intNullableList: [],
+            nullableIntList: [],
+            nullableIntNullableList: nil)
+
+        let updatedContainer = ListIntContainer(id: container.id,
+                                                test: 2,
+                                                nullableInt: nil,
+                                                intList: [1, 2, 3],
+                                                intNullableList: [1, 2, 3],
+                                                nullableIntList: [1, 2, 3],
+                                                nullableIntNullableList: [1, 2, 3])
+
+        guard case .success(let createdModel) = mutateModel(request: .create(container)),
+              case .success(let updatedModel) = mutateModel(request: .update(updatedContainer)),
+              case .success(let queriedModel) = queryModel(request: .get(ListIntContainer.self, byId: container.id)),
+              case .success(let deletedModel) = mutateModel(request: .delete(updatedContainer)),
+              case .success(let emptyModel) =
+                queryModel(request: .get(ListIntContainer.self, byId: container.id)) else {
+            XCTFail("Failed to test CRUD operations")
+            return
+        }
+        XCTAssertEqual(createdModel, container)
+        XCTAssertEqual(updatedModel, updatedContainer)
+        XCTAssertEqual(queriedModel, updatedContainer)
+        XCTAssertEqual(deletedModel, updatedContainer)
+        XCTAssertNil(emptyModel)
+    }
+
+    func testListStringContainer() throws {
+        let container = ListStringContainer(
+            test: "test",
+            nullableString: nil,
+            stringList: ["value1"],
+            stringNullableList: [],
+            nullableStringList: [],
+            nullableStringNullableList: nil)
+
+        let updatedContainer = ListStringContainer(id: container.id,
+                                                   test: "test",
+                                                   nullableString: "test",
+                                                   stringList: ["value1"],
+                                                   stringNullableList: ["value1"],
```

Comment:
should we also test when these hold `nil` if/where possible?

Reply:
yes that sounds good, adding `testListContainerWithNil` test for APi and DataStore